src/Components/ArticlePage/Article.jsx

```diff
@@ -1,15 +1,51 @@
-import { convertTopic, convertDate } from "../../utils/utils";
+import { capitaliseFirstLetter, convertDate, addAdjective } from "../../utils/utils";
+import { useState } from "react";
+import { updateArticleVotes } from "../../api";
 
 export const Article = ({ article }) => {
-  const { title, topic, author, body, created_at, votes, article_img_url } =
-    article;
+  const {
+    article_id,
+    title,
+    topic,
+    author,
+    body,
+    created_at,
+    votes,
+    article_img_url,
+  } = article;
 
-  const topicName = convertTopic(topic);
+  const topicName = capitaliseFirstLetter(topic);
   const date = convertDate(created_at);
+  const [votesAdded, setVotesAdded] = useState(0);
+  const [error, setError] = useState(null);
+
+  const handleUpvote = () => {
+    setVotesAdded((currVotes) => {
+      return currVotes + 1;
+    });
+    setError(null);
+    updateArticleVotes(article_id, 1).catch((err) => {
+      setLikesCount((currentLikesCount) => currentLikesCount - 1);
+      setError("Your upvote was not successful. Please try again!");
+    });
+  };
+  const handleDownvote = () => {
+    setVotesAdded((currVotes) => {
+      return currVotes - 1;
+    });
+    setError(null);
+    updateArticleVotes(article_id, -1).catch((err) => {
+      setLikesCount((currentLikesCount) => currentLikesCount - 1);
+      setError("Your downvote was not successful. Please try again!");
+    });
+  };
```

Comment:
These two functions are almost identical, can you think of a way to combine them into a single more dynamic function?